### DISCUSSION TITLE
Wrong filament routes

### DISCUSSION THREAD

**Discussion by dev-dev-15** [filamentphp/filament#8295] [Help] [answered]
I don't know if it's a bug in Filament, or it's because of misconfiguring Nginx so I just wanted to share it here.
So I have two Laravel applications each one has it's own database, running on the same subdomain:
1. The first one is on: app1.domain.local
2. The second one is on app1.domain.local/app2

Here is the Nginx configuration:
```shell
server {
    listen 80;
    server_name app1.domain.local;
    root "S:/laragon/www/app1/public";
    
    index index.html index.htm index.php;
 
    location / {
        try_files $uri $uri/ /index.php$is_args$args;
		autoindex on;
    }

	location /app2 {
        alias "S:/laragon/www/app2/public";
        try_files $uri $uri/ @test;

        location ~ \.php$ {
            include snippets/fastcgi-php.conf;
            fastcgi_pass php_upstream;
            fastcgi_param SCRIPT_FILENAME $request_filename;		
            #fastcgi_pass unix:/run/php/php7.0-fpm.sock;
        }
    }

    location @test {
        rewrite /test/(.*)$ /test/index.php?/$1 last;
    }
    
    location ~ \.php$ {
        include snippets/fastcgi-php.conf;
        fastcgi_pass php_upstream;		
        #fastcgi_pass unix:/run/php/php7.0-fpm.sock;
    }
	
	
    charset utf-8;
	
    location = /favicon.ico { access_log off; log_not_found off; }
    location = /robots.txt  { access_log off; log_not_found off; }
    location ~ /\.ht {
        deny all;
    }
}

``` 

On the first app, there is no problem at all.
On the second app which is at /app2, weird things happen.
It's working just fine, it's showing data from its own database, but when I create a new record, when I click on the submit button the live wire post request goes to `http://app1.domain.local/livewire/update`, shouldn't it go to `http://app1.domain.com/TEST/livewire/update`? and the surprise is that the record is created in the database of the first app, and the redirection after creation goes to the first app too, so basically, since I clicked on the submit button to create a record it's jumped to the first app.

Routes also default to the first app, even if I call them in the second app, I have a button for example that redirects the user to a resource create page:
```php
public function createCustomer()
    {
        return redirect(route('filament.dashboard.resources.customers.create'));
    }
``` 

When I call this one from the second app, it is expected to take me to `app1.domain.local/TEST/dashboard/customers/create`, but goes to  `app1.domain.local/dashboard/customers/create`

**Comment by leandrocfe** [comment]
You can create `app2.domain.local` (app2/public) to avoid conflicts

**Reply by dev-dev-15** [reply]
Unfortunately, that's not an option, it has to be app1.domain.local/app2.

**Reply by leandrocfe** [reply]
Maybe [this post](https://webdock.io/en/docs/how-guides/laravel-guides/multiple-laravel-installs-subfolders-nginx-rewrite-rules-full-guide) can help you

**Reply by dev-dev-15** [reply]
thanks

**Comment by theHocineSaad** [comment, marked as answer]
The issue is with **Livewire**, **not Filament** (Filament is built using Livewire).
By default, Livewire sends all of its requests to `domain.com/livewire/update`, even if your app is in a subroute, e.g.: `domain.com/app`.
It is possible to override that endpoint using (you can add that route to your routes file):
```php
Livewire::setUpdateRoute(function ($handle) {
    return Route::post('/app/livewire/update', $handle);
});
``` 
Now the requests will go to `domain.com/app/livewire/update`.

Source: https://livewire.laravel.com/docs/installation#configuring-livewires-update-endpoint

**Reply by dev-dev-15** [reply]
that's the problem, thanks
